init.lua

```diff
@@ -76,6 +76,44 @@ minetest.register_globalstep(function(dtime)
     end
 end)
 
+--special keys
+minetest.register_on_mods_loaded(function()
+    for name, def in pairs(minetest.registered_nodes) do
+        local old_node_drop = def.on_drop
+        local on_drop = function(itemstack, dropper, ...)
+            for _, callback in pairs(controls.registered_on_press) do
+                callback(dropper, "_drop")
+            end
+            return old_node_drop(itemstack, dropper, unpack({...}))
```

Comment:
This should be just `...` instead of `unpack({...})`. The latter handles `nil`s in `...` incorrectly (and is less efficient).